pkg/handler/enabled_flag_validation.go

```diff
@@ -0,0 +1,201 @@
+// args = 1,2,/\d+/
+// putting it in regex, array order does not flatten it
+// args = /\d+/, [1,2]
+//
+// Govaluate function to evaluate an embedded govaluate expression (the first argument, a string)
+// for an array of 'items' (which should be a map[string]interface{}),
+// and if the subexpression returns true for any of the sub items
+// then the Any expression also returns true. It will early out once
+// it finds a single true result.
+func Any(args ...interface{}) (interface{}, error) {
+	if len(args) < 2 {
+		return false, fmt.Errorf("Expected 2+ arguments, but got %d", len(args))
+	}
+
+	function, ok := args[0].(string)
+	if !ok {
+		return false, fmt.Errorf("arg 0 is not a string!")
+	}
+
+	arr, ok := args[1].([]interface{})
+	if !ok {
+		return false, fmt.Errorf("arg 1 is not a slice! ***%v*** %v", args, reflect.TypeOf(args[1]))
+	}
+
+	functions := map[string]govaluate.ExpressionFunction{
+		"regexMatch": RegexMatchFunc,
+	}
+	expression, err := govaluate.NewEvaluableExpressionWithFunctions(function, functions)
+	if err != nil {
+		return false, err
+	}
+
+	found := false
+	for _, item := range arr {
+		params, ok := item.(map[string]interface{})
+		if !ok {
+			return false, fmt.Errorf("item is not a map %v", item)
+		}
+
+		result, err := expression.Evaluate(params)
+		if err != nil {
+			return false, err
+		}
+
+		if boolResult, ok := result.(bool); ok {
+			if boolResult {
+				found = true
+				break
+			}
+		}
+	}
+	return found, nil
+}
+
+func getEvaluationRules() ([]*govaluate.EvaluableExpression, error) {
+	createEvaluationRule.Do(func() {
+		for _, structExpressionString := range config.Config.EnabledFlagValidationRules {
+			functions := map[string]govaluate.ExpressionFunction{
+				"regexMatch": RegexMatchFunc,
+				"any":        Any,
+			}
+
+			expression, err := govaluate.NewEvaluableExpressionWithFunctions(structExpressionString, functions)
+			if expression == nil || err != nil {
+				evaluationRulesSetupError = err
+				return
+			}
+			evaluationRules = append(evaluationRules, expression)
+		}
+	})
+
+	return evaluationRules, evaluationRulesSetupError
+}
+
+func validateEnabledFlag(f *entity.Flag) (valid bool, err error) {
+	if config.Config.EnabledFlagValidationOperation == config.FlagValidationOperationAND {
+		valid = true
+	}
+
+	evaluationRules, err := getEvaluationRules()
+	if err != nil {
+		return false, err
+	}
+
+	for _, validationRule := range evaluationRules {
+		params := structs.Map(f)
+
+		result, err := validationRule.Evaluate(params)
+		if err != nil {
+			return false, err
+		}
+
+		resultBool, ok := result.(bool)
+		if !ok {
+			return false, fmt.Errorf("Unknown evaluation rule result type (%v)", result)
+		}
+
+		if config.Config.EnabledFlagValidationOperation == config.FlagValidationOperationOR {
+			valid = valid || resultBool
+			if valid {
+				break
+			}
+		} else if config.Config.EnabledFlagValidationOperation == config.FlagValidationOperationAND {
+			valid = valid && resultBool
```

Comment:
Perhaps not too consequential, but I think we can early exit if any validation operation fails.